### ISSUE TITLE
The scale is not correct

### ISSUE THREAD

**Issue by lliuz** [ClubAI/MonoDepth-PyTorch#7]
https://github.com/ClubAI/MonoDepth-PyTorch/blob/50e50b6e64080c0c05076e8552dde17342e5b236/loss.py#L44-L45

I think this should be [-0.5, 0.5] instead of [-1, 1]. which results in a 2x reduction in depth but has no influence with disparity.

**Comment by askorikov** [comment]
The thing is that PyTorch's `grid_sample` assumes the coordinates to be normalized between -1 and 1. See [docs](https://pytorch.org/docs/stable/nn.html?highlight=grid_sample#torch.nn.functional.grid_sample)

**Comment by lliuz** [comment]
Sorry, you're right. but I still don't know why the depth value has indeed shrunk twice.

**Comment by lliuz** [comment]
I got it. It's my fault that I forget to set `vel_depth` in [the `generate_depth_map` function](https://github.com/mrharicot/monodepth/blob/master/utils/evaluation_utils.py#L169), so I get the wrong ground truth.
Nice work, thanks for your prompt reply!

**Comment by askorikov** [comment]
No problem. Good luck!

**Comment by lliuz** [comment]
I still have a doubt. `x_shift` is the disparity which should be `0~1`. Although it is limited to `0~0.3` same to the original tensorflow version, the value is normalized with the width of the image. The `grid_sample` in this repo adopts `-1, -1` as the left top pixel and `1, 1` as the right bottom, so we should times 2 to `x_shift`( for example, in the current version, for the `-1, -1`, with shift 1 the sampler will get the pixel in the center instead of the right top). 

https://github.com/ClubAI/MonoDepth-PyTorch/blob/50e50b6e64080c0c05076e8552dde17342e5b236/loss.py#L49

I think we should modify this line to
```
flow_field = torch.stack((x_base + 2*x_shifts, y_base), dim=3)
```
Besides, we can modify the [get_disp](https://github.com/ClubAI/MonoDepth-PyTorch/blob/50e50b6e64080c0c05076e8552dde17342e5b236/models_resnet.py#L138) to
```
return 0.15 * F.sigmoid(x)
```
to keep same to the tensorflow repo.

**Comment by askorikov** [comment]
Oh, indeed. I do agree with shifts correction. This should be fixed by a20d70549d44fbe42e43b58d33e82420f6075d66 (note that I decided to first do everything in [0, 1] coordinates and then properly transform the flow field while passing to `grid_sample`).

However, I think `get_disp` should be okay after this fix - now we assume all shifts to lie in [0, 1] and truncate them to 0.3 the same way it's done in the original repo.

**Comment by lliuz** [comment]
The problem has been solved, thanks again. @5ht2
